### ISSUE TITLE
Pivot accessor not resolved as magic property on `BelongsToMany` relations

### ISSUE THREAD

**Issue by alies-dev** [psalm/psalm-plugin-laravel#504]
## Description

When accessing `$model->pivot` on a model retrieved through a `BelongsToMany` or `MorphToMany` relationship, Laravel provides a magic `pivot` property (an instance of `Illuminate\Database\Eloquent\Relations\Pivot` or a custom pivot class). The plugin does not recognize this accessor, causing `UndefinedMagicPropertyFetch` at Psalm errorLevel 1.

## Impact

**5 false positives** across [monicahq/monica](https://github.com/monicahq/monica) (262 project files).

Affected patterns:
- `App\Models\Module::$pivot` (2 occurrences)
- `App\Models\Address::$pivot` (1)
- `App\Models\ContactReminder::$pivot` (1)
- `App\Models\LifeMetric::$pivot` (1)

## Reproduction

```php
$user = User::with('roles')->first();

foreach ($user->roles as $role) {
    // Triggers UndefinedMagicPropertyFetch
    echo $role->pivot->created_at;
}
```

## Expected Behavior

The plugin should recognize `$model->pivot` as a valid magic property. The type depends on the relationship definition:

- Default: `Illuminate\Database\Eloquent\Relations\Pivot`
- Custom pivot: when the relationship uses `->using(CustomPivot::class)`, the type should be the custom pivot class

Laravel also supports the `$model->getRelationValue('pivot')` alias and custom accessor names via `->as('subscription')` (e.g., `$model->subscription` instead of `$model->pivot`).

## Context

Found while auditing [monicahq/monica](https://github.com/monicahq/monica). After #502 (relationship accessor resolution), these are among the remaining Eloquent-related `UndefinedMagicPropertyFetch` errors.

**Comment by alies-dev** [comment]
Closing as duplicate of #641, which has the full design proposal (using()/withPivot()/as() resolution). Monica impact data preserved in a comment there.
